crates/proof-of-sql/src/proof_primitive/dory/extended_dory_inner_product.rs

```diff
@@ -34,6 +34,7 @@ pub fn extended_dory_inner_product_verify(
     transcript: &mut Transcript,
     mut state: ExtendedVerifierState,
     setup: &VerifierSetup,
+    fold_s_tensors_verify: fn(&ExtendedVerifierState) -> (F, F),
```

Comment:
NIT: for functions as parameters, using a trait bound instead of a fn pointer is a little more flexible for other use cases (like closures)
```suggestion
    fold_s_tensors_verify: impl Fn(&ExtendedVerifierState) -> (F, F),
```